### PR TITLE
SJ - PREVENT EMPTY PARSER VERSION NAME: As Allister I want version names/messages to be validated, so that empty names are not allowed

### PR DESCRIPTION
**Pull request by Reinkaos**
Acceptance Criteria

Harvest operators are not able to save empty parser version names
Harvest operator is informed that the name/message field is not allowed to be empty

### CODE REVIEW

File: spec/models/version_spec.rb [hunk 1]
```diff
@@ -8,13 +8,32 @@ class Program
...
   before do
     version.versionable = program
     version.user = build(:user, email: 'test@test.co.nz')
   end
 
+  describe 'validations' do
+    context 'Invalid Version' do
+      let(:invalid_version) { build(:version, message: nil) }
+
+      it 'requires message to be valid' do
```
Comment (ostigley) [inline spec/models/version_spec.rb:22]: perhaps `it 'requires message to be present' do`

File: app/views/parsers/_form.html.erb [hunk 2]
```diff
@@ -2,8 +2,12 @@
   <% if can? :update, @parser %>
     <%= f.text_area :content, class: "code-editor read" %>
 
+    <% if @parser.errors.any? %>
+      <span class="error">The parser version message is required.</span>
+    <% end %>
```
Comment (paul-mesnilgrente) [inline app/views/parsers/_form.html.erb:7]: The error display is already handled in the `custom_form_builder.rb`. Was it not showing up?

Comment (Reinkaos) [reply app/views/parsers/_form.html.erb:7]: No it wasn't showing up. It could be because the version is a relation of the parser?

**Comment by paul-mesnilgrente** [comment]
I believe the Parser model is not the only one using this Version concern. Should we add this front-end validation on the other models?

**Comment by Reinkaos** [comment]
Ah you're right, it's also on the snippet. I'll update now

File: app/views/parsers/_form.html.erb (hunk 2 above)
Comment (paul-mesnilgrente) [reply app/views/parsers/_form.html.erb:7]: Could be, the error has to be `errors[:message]` or `errors[:message_id]`. Makes me think that you are displaying `The parser version message is required.` if there's any error. Could you display in just if that's the error you have? I believe you can have other kind of errors on the parser.

Comment (Reinkaos) [reply app/views/parsers/_form.html.erb:7]: Yeah sure, the other errors for the parser version are pre populated so it should be safe. I'll see why the custom form errors aren't coming through

**Pull request merged**